pkg/utils/utils.go

```diff
@@ -192,30 +193,30 @@ func InsertApplicationProfileContainer(profile *v1beta1.ApplicationProfile, cont
 	}
 }
 
-func (watchedContainer *WatchedContainerData) SetContainerType(wl workloadinterface.IWorkload, containerName string) {
+func (watchedContainer *WatchedContainerData) SetContainerInfo(wl workloadinterface.IWorkload, containerName string) {
+	checkContainers := func(containers []v1.Container, containerType ContainerType) {
+		var containerNames []string
+		for i, c := range containers {
+			containerNames = append(containerNames, c.Name)
+			if c.Name == containerName {
+				watchedContainer.ContainerIndex = i
+				watchedContainer.ContainerType = containerType
+				watchedContainer.ContainerNames = containerNames
+			}
+		}
+	}
+	// containers
 	containers, err := wl.GetContainers()
 	if err != nil {
 		return
 	}
-	for i, c := range containers {
-		if c.Name == containerName {
-			watchedContainer.ContainerIndex = i
-			watchedContainer.ContainerType = Container
-			break
-		}
-	}
+	checkContainers(containers, Container)
 	// initContainers
 	initContainers, err := wl.GetInitContainers()
 	if err != nil {
 		return
 	}
-	for i, c := range initContainers {
-		if c.Name == containerName {
-			watchedContainer.ContainerIndex = i
-			watchedContainer.ContainerType = InitContainer
```

Comment:
TODO: check ephemeral containers

Reply:
Ah nevermind.. I see you merge to main, so we'll have to pull your changes to the feature branch